build/build-docker-deploy.sh

```diff
@@ -24,7 +30,7 @@ if [ "${1-}" = "docker" ]; then
 fi
 
 # Build the cockroach and test binaries.
-$(dirname $0)/builder.sh $0 docker
+$(dirname $0)/builder.sh $(path_for_docker) docker
```

Comment:
If `build-docker-deploy.sh` is launched with `./build-docker-deploy.sh` from  `cockroach/build` directory. $0 will be `./build-docker-deploy.sh` which does not works since workdir for the docker image is mapped to `cockroach` directory. `build-docker-deploy.sh` exists in `cockroach/build` directory not in `cockroach` directory.